coreos-config.yaml.tpl

```diff
@@ -0,0 +1,34 @@
+#cloud-config
+
+coreos:
+  etcd2:
+    discovery: ${etcd_discovery_url}
+    advertise-client-urls: http://$private_ipv4:2379
+    initial-advertise-peer-urls: http://$private_ipv4:2380
+    listen-client-urls: http://0.0.0.0:2379
+    listen-peer-urls: http://$private_ipv4:2380
+
+  units:
+    - name: etcd2.service
+      command: start
+    - name: docker-tcp.socket
+      command: start
+      enable: true
+      content: |
+        [Unit]
+        Description=Docker Socket for the API
+
+        [Socket]
+        ListenStream=4243
+        Service=docker.service
+        BindIPv6Only=both
+
+        [Install]
+        WantedBy=sockets.target
+
+    - name: docker.service
+      drop-ins:
+        - name: 50-insecure-registry.conf
+          content: |
+            [Service]
+            Environment=DOCKER_OPTS='--insecure-registry="${docker_regisry_internal_ip}/32"'
```

Comment:
We should use a hostname here instead of an IP address so that we don't regress [#93690844](https://www.pivotaltracker.com/story/show/93690844).